<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>PSSI Salmon Science Speaker Series</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./index.html">Home</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">PSSI Salmon Science Speaker Series</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Projects</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
 <span class="menu-text">pages/*.qmd</span>
  </li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#salmon-science-projects" id="toc-salmon-science-projects" class="nav-link active" data-scroll-target="#salmon-science-projects">üß™ Salmon Science Projects</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">PSSI Salmon Science Speaker Series</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="salmon-science-projects" class="level2">
<h2 class="anchored" data-anchor-id="salmon-science-projects">üß™ Salmon Science Projects</h2>
<p>Welcome to the gallery of research projects. Click on any card to explore the details.</p>
<div class="gallery">
<div class="gallery-card">
<p><a href="./pages/2394.html"><strong>Enhanced salmon bycatch monitoring and sampling in the Pacific region groundfish trawl fishery</strong></a><em>Groundfish trawls (non-salmon) are prohibited from selling or trading salmon bycatch, yet these commercial fisheries catch large numbers of salmon without management measures. This harvest sampling study counts salmon bycatch by species and collects tags (coded wire tags) and DNA from Chinook to estimate the total catch rate by stock, especially stocks of concern.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2397.html"><strong>Identifying good practices for considering Indigenous Knowledge in Rebuilding Plan targets and stock assessment reference poi</strong></a><em>Indigenous knowledge includes countless generations of place-based Pacific salmon experience that cannot be recreated or measured. Guided by the Eptuaptmumk (‚ÄúTwo-Eyed Seeing‚Äù) framework, we are partnering with First Nations on the West coast of Vancouver Island (WCVI) and University of Victoria to identify rebuilding targets and reference points for Chinook salmon that pair Indigenous Knowledge Systems with Western Science, based on interviews and focus groups with Indigenous Knowledge holders and a review of assessment literature.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2398.html"><strong>WCVI Sediment Transport and Redd Scour Assessment</strong></a><em>Climate change influences freshwater flows and sediment movement during the salmon egg incubation stage. Frequent high-severity flows can impact egg survival rates due to redd scour events and sediment deposition. This research develops and assesses methods for measuring redd scour and sediment deposition that may impacts redds to inform salmon habitat restoration, enhancement, and rebuilding efforts.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2399.html"><strong>Risk Assessment Method for Salmon ‚Äì Development for Regional Implementation</strong></a><em>Researchers in 2010 developed a tool called the Risk Assessment Method for Salmon that uses population and ecosystem information to identify and prioritize salmon population bottlenecks at different life stages. The tool constantly evolves, but currently uses awkward technology and lacks marine information. This stock management support project updates regional salmon information including marine and freshwater life stage information and converts the old Excel program to a more visual and user-friendly R programming design.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2400.html"><strong>Assessment of SEP Chinook and Coho broodstock ELISA screening data by modelling for explanatory variables, and yearling DFAT prevalence data by modelling for predictive variables.</strong></a><em>Screening salmon brood stock for bacterial kidney disease has allowed hatcheries to reduce the disease in young salmon and returning adults. Hatcheries now have 16 years of disease test data along with genetic (parentage) information. This hatchery study collects disease data and genetics from Chinook and coho at the Spius, Puntledge, and Nitinat hatcheries to model variables that could improve hatchery methods.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2401.html"><strong>Epidemiological modeling of infectious hematopoietic necrosis virus in Sockeye salmon</strong></a><em>Long-term monitoring of the salmonid virus ‚Äúinfectious hematopoietic necrosis virus,‚Äù which can be deadly for sockeye, shows annual and geographic fluctuations between stocks, but for unknown reasons. This study examines IHNV prevalence across different watersheds, sockeye stocks, and environmental conditions in an effort to identify those factors responsible for driving IHNV prevalence and/or influencing disease within sockeye salmon populations.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2402.html"><strong>Quantifying Yukon Chinook migration mortality and its implications for fisheries management and rebuilding under the Fish Stock Provisions of the Fisheries Act</strong></a><em>Yukon River Chinook numbers have been declining and a questionable proportion of returning adults do not migrate past the border between Alaska and Canada. This transboundary tracking study uses radio-telemetry from towers and arial surveys to expand Alaska‚Äôs Chinook tagging efforts and locate the area and cause of the drop in numbers using death rate, migration speed, and timing.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2403.html"><strong>San Juan River Adult and Juvenile Assessment Program</strong></a><em>Pacheedaht First Nation (PFN) have extensive monitoring and restoration projects underway in the San Juan Watershed, but some data gaps for San Juan Chinook still exist, and more detailed biological sampling is required. DFO Salmon Stock Assessment and PFN developed a holistic lifecycle monitoring program for San Juan Chinook through operation of a freshwater rotary screw trap, marine purse seining, and microtrolling, allowing researchers to monitor Chinook across all life stages and compare traits of natural and hatchery origin. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2404.html"><strong>Improving baseline knowledge of environmental conditions in Vancouver Islands fjords through observations and modelling, with a focus on hypoxia dynamics, climate change, and the potential implications for Pacific salmon</strong></a><em>Low-oxygen (hypoxia) conditions have been observed near the surface in salmon migration inlets on west coast Vancouver Island. The ecosystem monitoring component of this study measures salinity, temperature, and oxygen at different depths in Clayoquot Sound fjords in collaboration with Indigenous groups to establish present-day baseline conditions. The modelling component helps understand the drivers of hypoxia and how they may change in the future. Will salmon have to face more or less hypoxic conditions while migrating?</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2405.html"><strong>Biological models to support prioritizing salmon stocks under future climates</strong></a><em>Predicting how salmon stocks respond to climate change is difficult because salmon life stages span diverse habitats seasonally. This modelling project studies the relationships between salmon survival and environmental data in freshwater and marine habitats, then applies those relationships to more specific stocks with less available information. By modelling these relationships across all salmon stocks, decision-makers are better able to predict the state of salmon stocks under a variety of climate change scenarios.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2406.html"><strong>Changing coastal productivity: Using sediment cores, water properties and archived plankton data to identify changes at the bottom of the food web in BC‚Äôs coastal waters</strong></a><em>Has climate change reduced the amount of plankton at the bottom of the food web in BC waters, reducing the carrying capacity for salmon? Have smaller phytoplankton replaced diatoms, lengthening the food chain and making food less nutritious for salmon? This study uses chemical markers in sediment cores and water samples to investigate food web changes in BC coastal inlets.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2407.html"><strong>Follow the Fish: An integrated Chinook salmon assessment and monitoring program for the West Coast of Vancouver Island</strong></a><em>Juvenile Chinook salmon utilize nearshore marine ecosystems along the West Coast during their first marine year. This monitoring project samples juvenile Chinook during year-round salmon surveys in estuarine and marine waters along the WCVI. Researchers are characterizing fish distribution and health (condition, diet, growth) to understanding what factors may be limiting early marine survival. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2408.html"><strong>Improvement, Expansion and Modernization of Salmonid Health Diagnostic Services For Optimizing Salmonid Hatchery Health Management.</strong></a><em>Salmon disease testing methods have greatly expanded in the last decade resulting in increased accuracy and decreased turnaround time; implementation of these tests will increase the Finfish Diagnostic Laboratory‚Äôs diagnostic capacity. This hatchery lab project collects tissue samples from broodstock and young salmon to confirm new methods are sensitive and specific to pathogens of interest. Once validated, these tests will be used by the Finfish Diagnostic Laboratory. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2409.html"><strong>Convergent tracks: a tagging study to quantify salmon predation by sea lions</strong></a><em>Steller sea lions are hypothesized to feed on large numbers of adult Fraser River sockeye salmon when the salmon migrate past Triangle Island, near northern Vancouver Island. This project tags sockeye salmon and Steller sea lions, and also analyzes sea lion diets, to determine if the area is a survival bottleneck.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2410.html"><strong>Improved decision making for salmon by understanding the threats of freshwater Aquatic Invasive Species both now and in the future</strong></a><em>Aquatic Invasive Species (AIS) continue to be introduced and expand their ranges aided by climate change, threatening biodiversity and Pacific salmon survival. This risk assessment study targets over 400 invasive species within or approaching five freshwater ecoregions in BC. Researchers developed a Non-Indigenous Species Screening Tool (NISST) that allows managers to prioritize the (AIS) threats when creating regulations and salmon recovery plans.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2412.html"><strong>Investigation of the impacts of singular and coinciding acute climate stressors on the nutritional quality of the pteropod Limacina helicina, a juvenile Pacific salmon dietary species</strong></a><em>Ocean warming and acidic upwelling changes the zooplankton food web for young salmon in the Northeast Pacific. This project measures the survival and nutritional value of pteropods (zooplankton) under different temperatures and acidic conditions in a lab. Researchers then compare lab findings to natural zooplankton conditions.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2413.html"><strong>Barkley Sound and Clayoquot Sound Krill Monitoring</strong></a><em>Krill, a type of zooplankton, are a critical component of the marine food webs and important part of juvenile salmon diets. This ecosystem monitoring program collects krill on a monthly basis in two key feeding areas for early marine juvenile salmon along the west coast of Vancouver Island. We are monitoring how the amount and nutritional value (fats, protein and total energy) of krill varies with season and ocean conditions (temperature, salinity, and oxygen). </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2414.html"><strong>South Coast freshwater ecological indicator pilot</strong></a><em>Streamflow, water quality, and aquatic insect information is useful when assessing salmon habitat at risk. This freshwater ecosystem monitoring project expands the range of data collection to include between five to 14 more Chinook habitat systems where researchers sample streamflow, temperature, dissolved oxygen, pH, and benthic macroinvertebrates.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2415.html"><strong>A Decision Support Framework for Identifying and Characterizing the Non-fishing Threats to Pacific Salmon</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2416.html"><strong>Monitoring and predicting the exposure of Pacific salmon to harmful algal biotoxins</strong></a><em>Pacific salmon in Barkley Sound may be exposed to high levels of harmful algal biotoxins in the fall. Biotoxins have been detected in gill tissue from Chinook salmon collected in Barkley Sound. We are comparing biotoxin levels with environmental conditions and indicators of fish health. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2417.html"><strong>Optimization of feeds used in the hatchery production of Pacific Salmon.</strong></a><em>Hatchery feeds often come from commercial sources focused on other fish species and may be nutritionally incomplete for Pacific salmon. This hatchery feed study develops new diets specific to Chinook and coho and tests the salmon health under different feed trials.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2418.html"><strong>Prediction of reproductive success of Chinook salmon based on Thiamine concentrations in returning adults.</strong></a><em>Thiamine (vitamin B1) deficiency in salmon is diet-related and could be tied to warming oceans and changing food webs. The nutrient deficiency lowers the survival rate for salmon fry and has been observed in Yukon and California Chinook, but hasn‚Äôt been studied well in BC. This fish health monitoring study gathers Chinook tissue samples from the Albion test fishery and hatcheries to measure thiamine, fatty acids, and other health parameters to help predict and monitor vulnerable stocks. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2419.html"><strong>Pilot skills study of new hatchery tray configuration.</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2421.html"><strong>Measurement of stress hormones in scales and its application for the identification of conditions causing chronic stress in Pacific Salmon.</strong></a><em>Chronic stress impacts fish health and can be measured in salmon by sampling hormone (cortisol) and enzyme deposits in scales. This lab and hatchery study gathers coho and Chinook scale samples from broodstock surveys to develop methods of measuring chronic and acute stress hormone levels that could be expanded to monitor salmon health responses to stressors including climate change. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2422.html"><strong>Development and application of laser ablation-inductively coupled plasma-mass spectrometry (LA-ICP-MS) to determine saltwater entry size of juvenile salmonids and track habitat usage.</strong></a><em>Advances in lab microchemistry (mass spectrometry) can be used to track juvenile salmon life stage history from ear bone (otolith) samples. Body size at saltwater entry and habitat use can be measured and compared to fish health and age. This project develops methods for ‚Äúlaser ablation-inductively coupled plasma-mass spectrometry‚Äù to test on juvenile Chinook sampled in the Follow the Fish program.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2423.html"><strong>Complementing British Columbia‚Äôs coasts longest Sea Surface Temperature and Salinity records with 21st century technology to monitor a changing climate.</strong></a><em>Continuous marine salmon habitat measurements are important for fisheries management. The BC Shore Station Program collects long-term sea surface temperature and salinity at twelve coastal stations, but the data are coarse and limited by weather and labour. This ecosystem monitoring project is testing and installing a minimum of four independent water quality logging stations spanning the coast to collect and transmit hourly real-time data posted online.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2424.html"><strong>Mechanistic modelling to link hydrology to juvenile salmon habitat quality and productivity</strong></a><em>Predicting how much water fish need is a key management challenge in the face of climate change and increased water demand. Hydraulic habitat models are widely used for instream flow management, but do not currently account for dynamic water temperature. This project leverages large-scale flow and temperature monitoring across the North Thompson to develop flow-habitat models for juvenile coho, which incorporate dynamic temperatures and habitat structure.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2425.html"><strong>Development of geospatial tools for large-scale freshwater salmon habitat assessment by the Freshwater Spatial Ecology Program</strong></a><em>Identifying individual and cumulative threats to salmon and their habitat is critical for effective management and restoration actions. This project estimates and maps nine human activities and landscape disturbance-based threats and four climate change-based threats for the stream network in the Fraser Basin. Results identify threat levels across salmon Conservation Units and the human activities and land uses that contributed most to threats.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2426.html"><strong>Sakinaw Sockeye Juvenile Research on Measures to Increase Marine Survival</strong></a><em>Marine survival is a limiting factor for the endangered Sakinaw Lake sockeye smolts who migrate past hundreds of harbour seals at Hodgson Islands in the Strait of Georgia every spring. This salmon tracking study traps and PIT tags up to 2,500 sockeye smolts at a smolt fence, then transports the sockeye by boat past the seal colony and compares the survival rate of transported sockeye to naturally migrating smolts.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2427.html"><strong>Salish Sea plankton and oceanography</strong></a><em>Zooplankton monitoring in the Strait of Georgia is important to understand the food web and juvenile salmon survival, but the current monthly sampling timing is too low for confident statistics and modelling. This food web monitoring study collects zooplankton and ocean conditions on bi-weekly surveys in the spring and fall to more confidently model salmon diet and health.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2430.html"><strong>Feasibility of Estimating Chilko River Smolt Abundance Using Upward- and Side-Looking SONAR Methods</strong></a><em>Chilko Lake sockeye smolts are counted each spring from a fish-counting weir, and these counts inform broader Fraser River fisheries management decisions. However, with earlier spring freshets due to the warming climate, crews must remove the weir more often before the smolts are counted. This salmon monitoring project tests upward-looking and side-looking SONAR units that can be used during high flows, and compares the results to the weir counts.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2432.html"><strong>Barkley Sound and Clayoquot Sound acoustic monitoring of salmon and salmon prey</strong></a><em>Acoustic tools can provide important information on salmon ecology (food web) and salmon survival in marine areas. These salmon habitat and abundance monitoring studies use multi-frequency echosounders from mooring stations and mobile units in West Coast Vancouver Island sounds and the Salish sea to measure aspects of salmon forage species (zooplankton, krill, and Pacific herring) and count juvenile and adult salmon along their migration.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2433.html"><strong>Integrated Salish Sea acoustic monitoring of salmon and salmon prey</strong></a><em>Acoustic tools can provide important information on salmon ecology (food web) and salmon survival in marine areas. These salmon habitat and abundance monitoring studies use multi-frequency echosounders from mooring stations and mobile units in West Coast Vancouver Island sounds and the Salish sea to measure aspects of salmon forage species (zooplankton, krill, and Pacific herring) and count juvenile and adult salmon along their migration.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2434.html"><strong>SHERLOCK assay for rapid genotyping applications</strong></a><em>Hatchery staff use genetic markers to tell the difference between summer and fall runs of Chinook in the Puntledge River to preserve genetic integrity. However, genetic tests can often be slow and only applied later. This salmon genetics pilot project assesses rapid, in-season testing where genetic marker information could be provided in under an hour. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2435.html"><strong>Intergenerational transfer and parental origins of DNA methylation variation in Coho and Chinook Salmon</strong></a><em>Chemical reactions in a salmon body decide which genes are expressed in a process called ‚ÄúDNA methylation.‚Äù This hatchery genetics study sequences methylation across the genome to compare between hatchery and natural origin to see if salmon offspring inherit DNA methylation from parents (epigenetics) and how this could inform hatchery breeding methods. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2436.html"><strong>Chum whole genome sequencing for improved stock delineation</strong></a><em>Genetic sequencing is an effective tool to identify salmon species by stock, but this method is only available where enough genetic information has been gathered and sufficient differentiation has been detected. South Coast chum salmon have not been sequenced intensely enough to fully describe population structure delineation, so this genome sequencing study gathers DNA from across the region to assess differences in genetic markers that could be used to maximize the resolution.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2437.html"><strong>Adaptive genetic variation and climate change resilience in Canadian Pacific salmon</strong></a><em>Pacific salmon genetic markers are regularly used to manage fisheries and hatcheries. While these genetic analysis efforts provide important stock management information, they could also be used to assess diversity of salmon species relative to habitat changes. This salmon genetic adaptation study gathers Chinook and sockeye DNA samples and compares genomes to environmental conditions and climate projections.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2439.html"><strong>Fournier Transfer Near Infrared (FT-NIR) Spectroscopy; its application for rapid fish age estimation within the Sclerochronology Lab at the Pacific Biological Station</strong></a><em>Neural Network Techniques and Deep Machine Learning are being applied to all facets of science including the age estimation of salmon. The Sclerochronology Lab (SCL) is taking advantage of new technologies to increase productivity and address staffing challenges. Age data is a critical component in the development of models assessing changes in the health and abundance of salmon species and stocks.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2440.html"><strong>Mat-37 Improvements to the evaluation of abundance and stock composition for Fraser River Juvenile Salmon</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2442.html"><strong>Graduate Research into 1) habitat use by juvenile Chinook Salmon in the Canadian Okanagan River and Lake system and 2) predation on juvenile salmon in the Canadian Okanagan River and Lake system</strong></a><em>Okanagan Chinook are endangered due to habitat destruction and over-fishing. Juvenile Chinook habitat use and risk from predators has not been well studied and may be key factors regarding climate change, invasive species, and restoration efforts. This population study hires graduate students to research juvenile Okanagan Chinook and track survival through tagging.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2443.html"><strong>Quantitative assessment of the impact of Smallmouth Bass suppression efforts in Cultus Lake and SMB emigration from Cultus Lake</strong></a><em>Introduced smallmouth bass pose a predation risk to sockeye in Cultus Lake. However, some anglers prefer fishing for smallmouth bass and strongly oppose actions to lower the bass population. This predator research study captures (electrofrishing), tags (PIT and acoustic transmitters), and biosamples smallmouth bass to assess population control methods and interviews anglers to develop a community plan. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2444.html"><strong>Assessment of the distribution and abundance of juvenile Chinook salmon in the lower Thompson River basin.</strong></a><em>Historic flooding in 2021 impacted salmon spawning and rearing habitat throughout the lower Thompson River and tributaries. This salmon population monitoring study counts juvenile Chinook (night snorkel surveys) in the lower Thompson mainstem and tributaries and compare to previous years to assess the population and habitat recovery.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2447.html"><strong>Innovative Ecosystem Based Approaches to identify cumulative stressors: Salmon Fit-Chips and eDNA</strong></a><em>Salmon experience a variety of physiological stressors and pathogens over migration which can be monitored non-lethally using Fit-Chip technology. Chinook salmon will expose themselves to potentially impactful stressors to avail feeding opportunities, enhancing vulnerability to cumulative stressors. eDNA data resolves habitat usage for Chinook salmon and demonstrating they associate with high biomass areas (prey availability), including aquaculture facilities, and avoid harmful algae. Models addressing the cumulative and synergistic interplay between factors will inform which levers under human control will result in strongest outcomes for salmon.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2448.html"><strong>Developing a proactive, modernized, holistic approach to ensure optimal health and condition of Hatchery Production</strong></a><em>A proactive holistic health monitoring approach based on high throughput molecular technologies has been validated for utility in two salmon enhancement hatcheries, and is now ready for minimally invasive application in hatcheries culturing conservation stocks. The technical approach revealed annual shifts in stress, infection, and timing of smolt readiness. Future monitoring of pathogens in hatchery source water is crucial to minimize infection potential and maximize post-release survival in stocks of conservation concern.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2449.html"><strong>A decision-support tool that considers harvest, hatchery, and habitat management levers to support implementation of the Fisheries Act for Pacific salmon</strong></a><em>Resource managers rely on decision modelling tool and programs to predict the outcomes from different actions. For salmon management, often these tools fail to include hatchery or habitat parameters. This decision-support project develops an open source, single-population prototype model that can be applied to West Coast Vancouver Island Chinook as a case study, including hatchery and habitat levers, as well as, uncertainty and risk.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2451.html"><strong>Relative reproductive success of hatchery- versus natural-origin salmon in Canadian integrated populations</strong></a><em>Salmon bred from a hatchery are often less fit in wild environments than naturally bred salmon. However, this situation can change using different hatchery methods. This genetic study compares DNA samples from returning hatchery Chinook to natural Chinook in Sarita River to see if hatchery methods improve the spawning success of hatchery Chinook.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2452.html"><strong>Genetic associations with age of return in male Canadian Chinook Salmon</strong></a><em>Jacks (males that return a year early) are a natural source of genetic diversity for Chinook, but too many jacks in a population can have limited breeding success and reduce the body size of the population. Hatchery breeding increases the number of jacks, and hatchery staff screen out jacks by body size, which is uncertain. This genetic monitoring study samples male Chinook tissues from the Chilliwack, Puntledge, and Qualicum rivers to sequence the genes, compare the y-chromosome haplotypes specific to jacks, and develop a rapid screening test for broodstock selection.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2453.html"><strong>Epigenetic variation between hatchery- and natural-origin Canadian Chinook Salmon</strong></a><em>There are few genetic differences between hatchery-bred and natural salmon in populations enhanced by the Salmonid Enhancement Program (SEP), but structural modifications to genes may affect gene expression and impact salmon fitness in the wild. This genetic sequencing study gathers Chinook samples from hatcheries at the Nicola, Quinsam, Sarita, Atnarko, and Chilliwack rivers to compare epigenetic (methylation) differences between hatchery and wild Chinook to better monitor the proportion of hatchery salmon in a river.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2454.html"><strong>Kim Hyatt‚Äôs Legacy Data: Juvenile Sockeye Biosample Data and Summary Metrics for B.C. Study Lakes, 1977-present</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2455.html"><strong>YTRA TBR Weather Stations</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2493.html"><strong>Improved understanding of cumulative impacts on salmon survival across freshwater life-stages; tools and approaches for mechanistic assessments</strong></a><em>Models of landscape-scope watershed assessments often can‚Äôt capture accurate salmon health and life-cycle dynamics because they lack the mechanistic tools to show cumulative impacts and multiple stressors in the environment. This salmon health and habitat (ecophysiology) monitoring project biosamples North Thompson juvenile coho and Fraser juvenile sockeye, along with water quality measurements to develop finer tools to model fish condition in relation to habitat change and climate conditions.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2494.html"><strong>Modeling interactions of environmental, biological and infectious factors with respect to production of Pacific salmon (Oncorhynchus spp.) at the Quinsam River hatchery</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2504.html"><strong>Identifying and characterizing tire-related chemical (6PPD-quinone) toxic hotspots in salmon habitat</strong></a><em>Automobile tires break down and leave the toxic chemical 6PPD-quinone on roadways. Rain events wash 6PPD-quinone into salmon-bearing streams, sometimes in amounts lethal to coho salmon. This contaminant monitoring study samples water quality from 10 new sites on salmon-bearing streams in urban areas of greater Vancouver, Squamish, and Vancouver Island to detect toxic amounts and compare to stormwater green infrastructure areas.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2513.html"><strong>Characterizing and monitoring priority contaminants of concern in WCVI juvenile salmon</strong></a><em>Recent studies found that West Coast Vancouver Island Chinook had some of the highest contaminant concentrations. This fish health monitoring project coordinates with the Follow the Fish program to sample for over 600 contaminants in juvenile Chinook to rank the results to prioritize mitigation actions.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2522.html"><strong>Explanation for the Collapse of the Commercial Chum Salmon Fishery in British Columbia</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2534.html"><strong>Understanding the impact of Arctic outflow winds on British Columbia fjords.</strong></a><em>Arctic air masses move from the interior to coastal inlets, cooling and adding oxygen to the marine habitat. Fjords where this cooling occurs can become a marine refuge for salmon. This project monitors climate conditions in BC mainland fjords to measure the impact on salmon habitat and study how these factors may impact other coastal inlets. </em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2539.html"><strong>Climate Downscaling for salmon conservation in Marine Ecosystems</strong></a><em>Ocean and climate conditions influence salmon numbers and health. While efforts have been made to model future climate conditions, these models are broad in scope which creates a scale gap when applying projections to salmon habitat. This modelling project downscales broader global climate models to include coastal environments and salmon species distribution.</em></p>
</div>
<div class="gallery-card">
<p><a href="./pages/2540.html"><strong>Modelling to Support Annual Fraser River Sockeye and Pink Salmon Forecasts</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2545.html"><strong>Genetics Stock Identification and Parentage Based Tagging of Pacific salmon</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2546.html"><strong>Fishery &amp; Assessment Data Section - Data Stewardship Unit</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2560.html"><strong>PSSI High Seas Salmon Research - West Coast Vancouver Island (and transition to ONE juvenile program)</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2561.html"><strong>State of Pacific Salmon Report 2024-2025</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/1261.html"><strong>Salmon Marine Interactions - SMI - COE Section</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2563.html"><strong>Salmon Scanner and composite status data sets</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2562.html"><strong>Climate Change Vulnerability Assessments</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2547.html"><strong>Fishery &amp; Assessment Data Section - Knowledge Mobilization Unit</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/2572.html"><strong>Assessing Salinity Alteration of Nursery Conditions for Salmon Smolts following Construction of Roberts Bank Terminal-2</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/3103.html"><strong>Drivers and functional consequences of hatchery epigenetic variation</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/3104.html"><strong>Establishing a feed assessment program in Pacific salmon enhancement hatcheries</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/3107.html"><strong>Scenario Planning: improving the adaptability of Pacific salmon hatchery planning and implementation to current and future climate changes</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/3119.html"><strong>Genomic consequences of stray hatchery introgression</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/3120.html"><strong>Genetic and epigenetic consequences of hatchery adaptation</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/3121.html"><strong>Impact of genetic divergence on outbreeding depression in Chinook Salmon</strong></a>**</p>
</div>
<div class="gallery-card">
<p><a href="./pages/3124.html"><strong>A data pipeline to support more rapid calculation of Proportionate Natural Influence</strong></a>**</p>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>